app/controllers/users.js

```diff
@@ -60,11 +63,32 @@ exports.signin = (req, res, next) => {
   const user = req.body ? req.body : {};
   emailValidation(req.body.email);
   userService.signin(user).then((signedInUser) => {
-    const token = sessionManager.encode({ username: signedInUser.email });
+    sessionManager.generateTokenAccess(signedInUser).then((token) => {
+      res.status(200);
+      res.set(sessionManager.HEADER_NAME, token);
+      res.end();
+    });
+  }).catch((err) => {
+    next(err);
+  });
+};
+
+exports.signout = (req, res, next) => {
+  userService.signout(req.user).then((signedOutUser) => {
     res.status(200);
-    res.set(sessionManager.HEADER_NAME, token);
     res.end();
   }).catch((err) => {
     next(err);
   });
 };
+
+exports.renew = (req, res, next) => {
+  const user = req.user;
+  const token = req.token;
+  if (req.body.renew_id === token.renewId) {
+    res.status(200);
+    res.send({ token });
```

Comment:
Here you are re-sending the old decoded token. By one hand the result should be encoded again so the frontend app receveis another jwt. And the idea of this endpoint is to send a new jwt with a new expirationDate so the session can keep alive.